wordpress-mobile/WordPress-iOS
# Gutenberg Opt-In

### Description:

Builds on top of @koke 's work in #10555 
Moves forward #10554 (still need to add the option to switch editors in the ellipsis menu)

### Details:

Changes:

- Deletes old code no longer needed.
- Splits `EditorSettings` into `GutenbergSettings` and `EditorFactory`.
- No longer showing the "New Editor coming up" alert when opening Gutenberg posts in Gutenberg.

### Testing:

Run the unit tests and make sure all is good.

**Test 1:**

1. Launch a fresh install of the app.
2. Make sure Gutenberg is disabled.
3. Enable Gutenberg.
4. Exit the App settings screen and go back to it.
5. Make sure Gutenberg is enabled.

**Test 2:**

1. Make sure Gutenberg is disabled.
2. Open a Gutenberg post with Aztec.
3. Make sure the "New Editor coming up alert" still shows up.

**Test 3:**

1. Make sure Gutenberg is enabled.
2. Open a Gutenberg post with Gutenberg.
3. Make sure the "New Editor coming up alert" is not showing up.
